src/main.rs

```diff
@@ -78,24 +104,66 @@ fn main() -> anyhow::Result<()> {
             url,
             buffer_address,
         } => {
-            let client = RpcClient::new(url);
-            let offset = UpgradeableLoaderState::size_of_buffer_metadata();
-            let account_data = client.get_account_data(&buffer_address)?[offset..].to_vec();
-            let program_hash = get_binary_hash(account_data);
-            println!("{}", program_hash);
+            let buffer_hash = get_buffer_hash(url, buffer_address)?;
+            println!("{}", buffer_hash);
             Ok(())
         }
         SubCommand::GetProgramHash { url, program_id } => {
-            let client = RpcClient::new(url);
-            let program_buffer =
-                Pubkey::find_program_address(&[program_id.as_ref()], &bpf_loader_upgradeable::id())
-                    .0;
-            let offset = UpgradeableLoaderState::size_of_programdata_metadata();
-            let account_data = client.get_account_data(&program_buffer)?[offset..].to_vec();
-            let program_hash = get_binary_hash(account_data);
+            let program_hash = get_program_hash(url, program_id)?;
             println!("{}", program_hash);
             Ok(())
         }
+        SubCommand::VerifyFromRepo {
+            solana_program_path,
+            repo_url,
+            program_id,
+            connection_url,
+            base_image,
+            name_of_program,
+            bpf_flag,
+        } => {
+            // Get source code from repo_url
+            let base_name = run_fun!(basename $repo_url)?;
+            let uuid = Uuid::new_v4().to_string();
+
+            run_fun!(git clone $repo_url /tmp/solana-verify/$uuid/$base_name)?;
+
+            // Get the absolute build path to the solana program directory to build inside docker
+            let build_path = PathBuf::from(format!("/tmp/solana-verify/{}/{}", uuid, base_name))
+                .join(solana_program_path.clone());
+            println!("Build path: {:?}", build_path);
+
+            // Build the code using the docker container
+            build(
+                Some(build_path.to_str().unwrap().to_string()),
+                base_image,
+                bpf_flag,
+            )?;
```

Comment:
I think /tmp/{uuid} should be deleted on any failure

Reply:
Makes sense, this crossed my mind as well. Included cleanup no matter the result in the latest commit. 

Also added:
- build by commit hash
- comments for flags